### ISSUE TITLE
Hard to read code when using Thoth.Json

### ISSUE THREAD

**Issue by lydell** [fsprojects/fantomas#685]
https://thoth-org.github.io/Thoth.Json/

I think this is very hard to read when the arguments to `map3` after the `fun` are bunched up to the right. At first glance it looks like they are part of the `| None ->` branch.

```fs
Decode.map3 (fun aggregateId event commitPayload ->
    match commitPayload with
    | Some payload ->
        Some
            { AggregateId = AggregateId aggregateId
              Event = event
              Payload = payload }
    | None -> None) (Decode.field "aggregate_id" Decode.string) (Decode.field "event" Decode.string) decodePayload
```

Something like this would be better:

```fs
Decode.map3
    (fun aggregateId event commitPayload ->
        match commitPayload with
        | Some payload ->
            Some
                { AggregateId = AggregateId aggregateId
                  Event = event
                  Payload = payload }
        | None -> None
    )
    (Decode.field "aggregate_id" Decode.string)
    (Decode.field "event" Decode.string)
    decodePayload
```

Thanks!

**Comment by nojaf** [comment]
Fantomas has as general rule if it fits on one line it should.
You can change the `PageWidth` to something smaller:
See:
https://jindraivanek.gitlab.io/fantomas-ui/#?fantomas=preview&code=CIUwxg9gJiB0C2BDADgZgAQAoBmBXAduogObEBOIxiALiAJJTogBuI+16k88AltQAqIAngBsIiRgFoAfAFgAUOiXok1MAAtOEbn0GjxjAO591C5egA+6AMraQ6ZMLET0Ms+eW34Idx48BvdABBUgoqWgZ0AF5g0MoaekYScniIqF8-DwBRVnZoplzqDMzlPWdGGMd9FwBfDKsAOQh8exl0JpaASixQSBhYbB4QEUYAImSwhIB9HihR9F7oOABnajIefGJuzEX+weGxljZqed2VtY2t9Bg+kDKDIA&config=N4IgCghg5gpg6gSwCYBcAWIBcBGADLgXyA
The result is somewhat smaller but not what you propose.

Another workaround could to add `//` at the end where you wish to force a line break.
However, I discovered #687 while trying that.

**Comment by nojaf** [comment]
Related https://github.com/fsprojects/fantomas/pull/1007/files#diff-a1b0b9a30fd94fe61b35a1a52ce27955R1514.
Maybe that check needs to be extended.

**Comment by nojaf** [comment]
Fixed as part of #1202.
[See example](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKARRckDoC2AhgA4DMYAFAGYCuAdmAQOaMBOijBALogJLxiIAbolqcwcPHgCWnAAoEAngBtYBfgFoAfAB0W2%2BhDCFO0ABbjYkmfOWr%2BAdxmnd%2Bw2AA%2BYAMqXEYIooqamBaLgZuED54iGERcWDAYACCzGwc3HxgALzJqexcvPxMrPkZ8LHxcQCiwqLZArWcFZVuNkH8OQG2wQC%2BzR5gAHKwtH5aQyOIAJQUmNiIOJRSiEr82pDFaQUA%2BlLlkGBzCAsAzpwsUrSMM%2BRHuEsra5BCIk0Hd6fnl9dgSPNtdkgABpwFATohOJxvidUGAANqgiCgOKQAAknAURGQaEgl04wMRhjRggISmo2PhhLcAAYQZVIDxaEhRF4pAAvZB0%2BIAFipYAAuoSelykXy0RisbDcaICSjUSSyRSEZUAIwiiKQACyBAAHgAZS6IPUiRicUyy%2BIqgBM1L5grcwsJyI16MxFMgACNYLAlBa3MTSeTYcr4lb1f6vIhpHAVLQkpwqkyAPKUA2jP0RSik8F2oXq53%2B12SnFen0ZiABxXBvkQEjholeALQRAAIUQlFgbHkLAI0W4LHLhnO5NzDvzYqL7tLvvrlaDaBDcW59YrjYIzbbHbYetg9kQLGgBHBjMEsEPUJGg4gw8Qo8MjrcBaJk6l08Hc6VNbAAFYV1A1xu7adogACqRBYgeR68LQp7nlIl5-lmSg5nE9r3uOcoSlO3ozhOCrzpSlQAGx-pAAGtkBbAAMJKEeJxUSMZwsNQ0CcJ2V5gEhKERGhEAPoYT4Vi%2BJY4e%2B8qBp%2BlQAOykeRm7AZqUYevuHFcbeqF5k6E5Ya%2BomzuJVYLl%2BAAcslNhRW6IAxsaqdm6k8Zpj7aW6ullvp%2BGSfEACcZnrogSSUP2DGSAQHE3nefEYS6OkiW5eESdWqq0l%2BZHmfJbCRtGPoIV%2BakRWA-GiphLmxbhcoeYllpqil5EBf2mVSDGOWVOFGljlpxXFlAb7uQlRmqmGKWMsynBJrQAAqLAKAA6k4tnIfZbi8QVUWFjF3V6fFhmEZadY1eZSQsLAdDwJgShSNI-Zhcxi2GMthUJM5XXSvivXbYuEQqsuKXajqPCUONpgiFUC1eKYnacLN8Bmhx3K2m16EddFJVQHiYkVf1lq-j9uqMksOpJhBXCdlUOpEGwJwnPBtAcd%2B8MOe1Tmde6aNvQRH1uCqJE4zqgyIPY52jHjUgE0TbEsKT5OIJT1ODNQeDKSwKY8NweAwn%2BKr5Q9glQMJUB4NQShQoLiAAGKdsY-bjS5bOeXEKoyT9hvG4aws6pqztSCb5ssJbKmkWYBA9qx%2B7bI40Pmlrq3PutL3o31O326ZPMAEpYJ28BQzDf5w1HSNrSjce25V9s%2Ban6csPAcsK-uyuq%2BrX6awjkX5zHhcG0bXuGj7fssNbkrF5jcQ2qRadwJXHud97FtcFdAemEH679mHeww3nTPI89rNbezX5WtV9K-YdPYKEmLAGmcWfmjn9NLY5AlPSzMqD4nERWoNh%2B6sfihnxfnDV4rOuUYG6qnXg-ZmUoO4u1GD3We%2B5%2B6ch3nbN%2Be16Tf1PufKQZxJ7QLNjPSE-sUqB2DsvcOa9m4rVbkJWO29yoJw5oYK031P46gAGoSRbJceA3wr6w1vnde%2BRVN5P1ekgkub9sYsNNnQVi1NOFMh4ava%2BX5c4UO1o-KUtCXQY1fm4K03MWHoFgJwAA4hCSWFMqYjF4X%2BOmYChEF2ej1MRQ836O0Pp7E2LYVDQAANYtmDr4iEJwxpeF7FZH08saaITsvYx6EDSrx3envZO9I%2BYC0NG2Tge4RAIMwEsWgMhqZJCZIpGuLAQHxDymo6O1DC7OLockyoVoy70gANKIEQEQf6gMRCMjCdENMiDKjVIZojDejiRFJN3pUEgyUWEg2kCcUwNiVH8IgPdWputY4NO0fQr8JAD7xDIt8JQ-kWCMHlq8aaiAPSamOPNbid9GbgOEa5MqeymnxBIB-Y5SRzqMFoNI2gsiRiskBVwagbBxqwGGq8Lg1NHm3Q2YI%2BJbzEkvwYbWVBfyjb7loAi4QO4rhlMVvky4RTGJIriTrcU9TNqNJmd85hxz0BYIIB6M5iysGmC8AoUQupqU1Kods%2BlcVGXILcCQSRxyvBfFYvcpAQqxl8VBPyLkuITimyprCPKPQgA).

**Comment by lydell** [comment]
Awesome! Thank you so much! 🥇
